content/en/docs/reference/labels-annotations-taints/_index.md

```diff
@@ -395,6 +395,19 @@ Used on: Service
 
 The control plane adds this label to an Endpoints object when the owning Service is headless.
 
+### service.kubernetes.io/topology-aware-hints {#servicekubernetesiotopology-aware-hints}
```

Comment:
Please use the updated terminology throughout:

1. The annotation should be `service.kubernetes.io/topology-mode`
2. The hints annotation is deprecated
3. The value should be "Auto" (note the capitalization)
4. Links should go to https://kubernetes.io/docs/concepts/services-networking/topology-aware-routing/
5. Domain-prefixed values are supported for custom/implementation-specific values per https://github.com/kubernetes/kubernetes/blob/99190634ab252604a4496882912ac328542d649d/staging/src/k8s.io/api/core/v1/annotation_key_constants.go#L154-L161

Reply:
In _this_ PR, we're documenting `service.kubernetes.io/topology-aware-hints` @robscott, as a fix for https://github.com/kubernetes/website/issues/36830.
Please review this as a PR to belatedly register and document **`service.kubernetes.io/topology-aware-hints`**.

PRs to document `service.kubernetes.io/topology-mode` are also welcome; I know that this is undocumented as of right now. Anyone is welcome to open a PR that documents the new annotation.

Reply:
`service.kubernetes.io/topology-aware-hints` was replaced by `service.kubernetes.io/topology-mode`, any reason we should document the old annotation? For reference, any use of the hints annotation will return a warning pointing to the new annotation: https://github.com/kubernetes/kubernetes/blob/99190634ab252604a4496882912ac328542d649d/pkg/api/service/warnings.go#L34

Reply:
We register (and document) all the annotations that we've ever used in official released code. Even if using them returns a warning, they should still be registered.

Reply:
(if we don't do that, we won't resolve issue #36830)